src/testplane.ts

```diff
@@ -101,7 +112,29 @@ export class Testplane extends BaseTestplane {
         );
         this.runner = runner;
 
-        this.on(MasterEvents.TEST_FAIL, () => this._fail()).on(MasterEvents.ERROR, (err: Error) => this.halt(err));
+        this.on(MasterEvents.TEST_FAIL, () => this._fail());
+        this.on(MasterEvents.TEST_FAIL, res => this._addFailedTest(res));
+        this.on(MasterEvents.ERROR, (err: Error) => this.halt(err));
+
+        if (runFailed) {
+            let previousRunFailedTests: FailedListItem[];
+            this.on(MasterEvents.INIT, async () => {
+                previousRunFailedTests = await this._readFailed();
+            });
+
+            this.on(MasterEvents.AFTER_TESTS_READ, testCollection => {
+                if (previousRunFailedTests.length) {
```

Comment:
I'm not sure if this is the best way to do this. In `src/test-reader/test-parser` we have this code:
```js
if (grep) {
    treeBuilder.addTestFilter(test => grep.test(test.fullTitle()));
}
```

We probably can implement this the same way, by adding a filter function.